### PR TITLE
155 provide example for the material mapping

### LINKED ISSUE #155: Provide example for the material mapping
**Is your feature request related to a problem? Please describe.**
Currently the assignment and mapping of material to geometry is only a theoretical concept. 

**Describe the solution you'd like**
Provide an example showcasing the mapping.

### PR DESCRIPTION
**Pull request by ClemensLinnhoff**
## Describe your changes

- [x] Add example glTF file showcasing both mesh-based and texture-based material assignment
- [x] Adjust material examples, mapping table, asset file etc. to fit to this example
- [x] Add Readme to explain the examples

## Issue ticket number and link

Fixes #155 

## Checklist before requesting a review

- [x] I have performed a self-review of my code/documentation.
- [x] My changes generate no new warnings during the documentation generation.

### CODE REVIEW

**Comment by ClemensLinnhoff** [comment]
@MatthiasThDs @ipg-jsc @ipg-sig @lyndyRott please also feel free to review this.

**Review by AsamDiegoSanchez — APPROVED** [review]
General review
Signed-off-by: Diego Sánchez <diego.sanchez@asam.net>

**Comment by lyndyRott** [comment]
looks like a good example to me. From the checker_xom.png (referring to red and blue material definitions) and the color-channels provided in the mapping table (referring to black and white texture names with red and blue rgb-values) it could be confusing to  understand the mapping, if the chosen color values are mixed channel. Nevertheless, I can understand that these are the chosen IDs for the material mapping.

**Comment by ClemensLinnhoff** [comment]
> looks like a good example to me. From the checker_xom.png (referring to red and blue material definitions) and the color-channels provided in the mapping table (referring to black and white texture names with red and blue rgb-values) it could be confusing to understand the mapping, if the chosen color values are mixed channel. Nevertheless, I can understand that these are the chosen IDs for the material mapping.

Thanks for your review. I can see how this might be confusing. But I actually deliberately chose other colors than black and white for the assignment texture to showcase, that these are just arbitrary mapping IDs and not the actual colors of the visual texture. I think we need to make this very clear in the documentation about the assignment/mapping.

**Comment by KimuraDIVP** [comment]
I think we'd better add the discription about assigning Matrial Mapping File (example_mapping.xomm) to Asset File (example_asset.xoma).

**Comment by ClemensLinnhoff** [comment]
> I think we'd better add the discription about assigning Matrial Mapping File (example_mapping.xomm) to Asset File (example_asset.xoma).

You are right, thanks for the remark. I somehow missed that.
I added it now as "materialMappingUri".

**Review by LudwigFriedmann — APPROVED** [review]
Very cool example! I was about to suggest adding an object with global material parameters in parallel to the textured cube, until I found the sphere in the code :)

File: examples/example_asset.xoma
[the remark this answers is not in the document]
Comment (ClemensLinnhoff) [reply examples/example_asset.xoma:1]: I added this.

File: examples/example_mapping.xomm [hunk 1]
```diff
@@ -11,9 +11,8 @@
     "creationDate": "20240703T101728Z",
   },
   "materialMapping": [
-    ["material_red", "materials/material_a.xomp", "metal with red paint"],
-    ["rgba:10;50;255;127", "materials/material_a.xomp", "metal with red paint"],
-    ["material_green", "materials/material_b.xomp", "metal with green paint"],
-    ["material_blue", "materials/material_c.xomp", "metal with blue paint"]
+    ["Material_Sphere", "example_material.xomp", "white aluminum"],
+    ["rgba:255;0;0;255", "example_material.xomp", "white aluminum"],
```
[the remark this answers is not in the document]
Comment (ClemensLinnhoff) [reply examples/example_mapping.xomm]: Good idea. I removed the alpha channel in the example.
If we are not using the alpha channel, a 24 Bit RGB texture would be enough, right?

[the remark this answers is not in the document]
Comment (MatthiasThDs) [reply examples/example_mapping.xomm]: Yes. If we want to allow the user to store in 24bit now, then I would say to not define it. Personally I would do everything in 32 Bit and later potentially use the Alpha channel, without the need to convert from 24 to 32 Bit first. But no strong opinion from my side.

**Review by MatthiasThDs — APPROVED** [review]

File: examples/example_mapping.xomm (hunk 1 above)
[the remark this answers is not in the document]
Comment (ClemensLinnhoff) [reply examples/example_mapping.xomm]: Decision: Make it normative to always use 32 bit textures, but currently only the RGB values are considered (first 24 bits). The alpha channel (last 8 bits) are reserved for features in future releases.

**Pull request merged**